kratos/processes/compute_nodal_gradient_process.h

```diff
@@ -46,10 +46,10 @@ namespace Kratos
 ///@name  Enum's
 ///@{
 
-    #if !defined(HISTORICAL_VALUES)
-    #define HISTORICAL_VALUES
-        enum HistoricalValues {Historical = 0, NonHistorical = 1};
-    #endif
+#if !defined(HISTORICAL_VALUES)
```

Comment:
Why is this needed?

Reply:
To know if the variable is historical FastGetSolutionStepValue or not GetValue

Reply:
Doing it at compile time is not a good idea. If you want do it as a runtime option...

Reply:
This is a definition of a enum, the precompiler code is just to avoid duplications in the definition

Reply:
this is macro magic, which is best avoided. Where else is this defined??